flaky test: [node/child_process] child_process.fork

This test appears to be flaky (and when it fails it causes the following test to fail as well):

https://github.com/denoland/deno_std/actions/runs/3202272796/jobs/5231090727

```
[node/child_process] child_process.fork => ./node/child_process_test.ts:511:6
error: AssertionError: Test case is leaking async ops.

 - 1 async operation to op_read was started in this test, but never completed.

To get more details where ops were leaked, run again with --trace-ops flag.
    at assert (deno:ext/web/00_infra.js:295:13)
    at asyncOpSanitizer (deno:runtime/js/40_testing.js:225:13)
    at async resourceSanitizer (deno:runtime/js/40_testing.js:370:7)
    at async Object.exitSanitizer [as fn] (deno:runtime/js/40_testing.js:427:9)
    at async runTest (deno:runtime/js/40_testing.js:833:7)
    at async runTests (deno:runtime/js/40_testing.js:1088:22)

[node/child_process execFileSync] 'inherit' stdout and stderr => ./node/child_process_test.ts:536:6
error: AssertionError: Test case is leaking async ops.
```

--- a/node/child_process_test.ts
+++ b/node/child_process_test.ts
@@ -524,7 +524,7 @@ Deno.test({
     const p = deferred();
     const cp = CP.fork(script, [], { cwd: testdataDir, stdio: "pipe" });
     let output = "";
-    cp.on("exit", () => p.resolve());
+    cp.on("close", () => p.resolve());
     cp.stdout?.on("data", (data) => {
       output += data;
     });

